Fix: issue 10611 add copy id feature to Panoramax

avijitdas126 opened the pull request:
### Changes
- Added "Copy ID" button/icon to relevant UI components.
- Implemented clipboard functionality to copy the ID with a single click.

### Benefits
- Improves user experience by simplifying ID retrieval.
- Aligns Panoramax features with those of Mapillary for consistency.

### Reference
Closes #10611

avijitdas126 committed:
Fix: issue 10611 add copy id feature to Panoramax

avijitdas126 commented:
Hello everyone,

I’ve implemented the "Copy ID" feature inspired by Mapillary as part of issue #10611. The current implementation adds a button/icon to easily copy the unique identifier of entities, improving the user experience.

I’d love feedback or suggestions to refine this feature further. If there are any specific improvements, alternative approaches, or edge cases you think we should consider, please feel free to share your thoughts.

Looking forward to collaborating with the community to make this feature as robust and useful as possible!

Thank you!

gedankenstuecke commented:
While not being a maintainer, I don't think this actually closes what the maintainers suggested to be the solution in #10611 as the PR as-is leads to a degradation of the workflow for mapillary. 

The suggestion was to copy the feature that was introduced for mapillary (c.f. https://github.com/openstreetmap/iD/issues/10611#issuecomment-2553846040), which allows adding the image ID as a new key/value pair to the features, instead of just copying the image ID to the clipboard. Could you try to expand on that mapillary feature to instead make it work for the other image providers too?

avijitdas126 commented:
Thanks for the feedback! I’ve updated the PR to ensure that image IDs are
added as key/value pairs for all relevant image providers, not just
Mapillary. Let me know if there are any further improvements needed.

On Thu, 6 Feb 2025, 7:14 pm Bastian Greshake Tzovaras, <
***@***.***> wrote:

> While not being a maintainer, I don't think this actually closes what the
> maintainers suggested to be the solution in #10611
> <https://github.com/openstreetmap/iD/issues/10611> as the PR as-is leads
> to a degradation of the workflow for mapillary.
>
> The suggestion was to copy the feature that was introduced for mapillary
> (c.f. #10611 (comment)
> <https://github.com/openstreetmap/iD/issues/10611#issuecomment-2553846040>),
> which allows adding the image ID as a new key/value pair to the features,
> instead of just copying the image ID to the clipboard. Could you try to
> expand on that mapillary feature to instead make it work for the other
> image providers too?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openstreetmap/iD/pull/10697#issuecomment-2639870331>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BBMNGLHFAPUXL4T46RNIIHT2ONRLPAVCNFSM6AAAAABVSRP64OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMZZHA3TAMZTGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

gedankenstuecke commented:
I don't think you've pushed any changes to this PR yet? 

Also: please check that the button for adding images doesn't overlap with existing UI, c.f. the high-res button for panoramax

![image](https://github.com/user-attachments/assets/98edad7c-6859-43ff-9240-39d667419774)

avijitdas126 commented:
I'll enhance the UI implementation and introduce the image ID as a new key/value pair to streamline the process.

tordans commented:
Pleas double check the hints that I posted in https://github.com/openstreetmap/iD/issues/10611#issuecomment-2553846040 and https://github.com/openstreetmap/iD/issues/10611#issuecomment-2575910888

tyrasd commented:
closing as stale
